src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineLinuxLauncher.java

```diff
@@ -127,12 +128,14 @@ protected void launch(ComputeEngineComputer computer, TaskListener listener, Ins
       scp.put(Jenkins.get().getJnlpJars("agent.jar").readFully(), "agent.jar", jenkinsDir);
 
       // Confirm Java is installed
-      if (!testCommand(computer, conn, "java -fullversion", logger, listener)) {
-        logWarning(computer, listener, "Java is not installed.");
+      String javaExecPath = node.getJavaExecPathOrDefault();
+      if (!testCommand(
+          computer, conn, String.format("%s -fullversion", javaExecPath), logger, listener)) {
+        logWarning(computer, listener, String.format("%s is not installed.", javaExecPath));
```

Comment:
Technically we're always looking for whether Java is installed, I think this would be better if changed to `Java is not installed at %s.`

Reply:
Nice catch.

Reply:
Done.